edx_oauth_client/middleware.py

```diff
@@ -46,14 +68,21 @@ def process_request(self, request):
 
         request.session[self.cookie_name] = auth_cookie
 
+        if reverse("logout") == current_url:
+            del request.session[self.cookie_name]
+            logout(request)
+            # FIXME: redirect is dirty fix to be able logout external portal.
+            # Please try to avoid and don't do it at edx side.
+            return redirect(urljoin(self.provider_url, self.provider_logout_url_path))
+
         if not is_same_user and is_auth:
             logout(request)
 
         if (auth_cookie and not is_continue and (not is_auth or not is_same_user)) or \
                 ('force_auth' in request.session and request.session.pop('force_auth')):
             query_dict = request.GET.copy()
             query_dict[REDIRECT_FIELD_NAME] = current_url
-            query_dict['auth_entry'] = 'login'
+            query_dict["auth_entry"] = "login"
```

Comment:
Why was this change added?

Reply:
As far as I know, in python double quotes are preferable to single quotes. I try to lead to this principle. If you do not mind, then in this module I will bring the quotes to a general view(double). Open to discussion.
Pushed a fix.